src/clj/election/controllers/elections.clj

```diff
@@ -59,15 +59,35 @@
   )
 )
 
+; RegExp from http://emailregex.com/
+(def email-pattern #"(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|\"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*\")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])")
+(defn is-valid-email? [value]
+  (re-matches email-pattern value))
+(defn get-invalid-mails [voters]
+  (filter
+    (comp not is-valid-email? second)
+    voters))
+
 (defn register-voters [{{election-id :election-id {voters-file :tempfile} :voters} :params {user :user} :session :as request}]
   (let [election (db/election (Integer. election-id))
     response (response/redirect (paths/election-path election-id))]
     (if (auth/can-register-voters? election user)
       (let [voters (voters-from voters-file)
-        added-voters-count (voters/register-voters (:id election) voters)]
-        (if added-voters-count
-          (assoc response :flash {:type :notice :message (i18n/t request :elections/new-voters-registered added-voters-count)})
-          (assoc response :flash {:type :error :message (i18n/t request :elections/voter-registration-failed)})
+            invalid-voters (get-invalid-mails voters)]
+        (if (empty? invalid-voters)
```

Comment:
Any chance we could make this report on the invalid ones but still add the valid ones?
I think it wouldn't be too hard

Reply:
Yes I can. My only concern that then, you would have to create a new file for the invalid voters spotted, as all other voters would have already been registered.
Currently, it just works or you have to retry with a valid list.
But I guess that it is pretty easy to create a new file with the corrected emails, so I will do as you suggest.

Reply:
The voter addition is actually idempotent so already registered voters are ignored. So you just need to fix the emails that were wrong.

Reply:
Perfect. I will do so.